draft-ietf-tls-dtls-rrc.md

```diff
@@ -177,6 +346,38 @@ address.
 {{path-challenge-reqs}} and {{path-response-reqs}} contain the requirements for
 the initiator and responder roles, broken down per protocol phase.
 
+# Enhanced Path Validation Procedure
+
+Note: This algorithm also takes the {{off-path}} scenario into account.
+
+The receiver that observes the peer's address or port update MUST stop sending
+any buffered application data (or limit the data sent to the unvalidated
+address to the anti-amplification limit) and initiate the return routability
+check that proceeds as follows:
+
+1. An unpredictable cookie is placed in a `return_routability_check` message of
+   type path_challenge;
+1. The message is sent to the previously valid address and a timer T (see
+   {{timer-choice}}) is started;
+1. The peer endpoint, after successfully verifying the received
+   `return_routability_check` message responds by echoing the cookie value in a
+   `return_routability_check` message of type path_response if the path through
+   which the message was received is preferred. If the path is not preferred,
+   a `return_routability_check` message of type path_delete MUST be returned;
+1. When the initiator receives and verifies the `return_routability_check`
+   message contains the sent cookie, it updates the peer address binding;
```

Comment:
the "updates the peer address binding" is for when the receiver gets back a verified path_response.  However, at this point it could have received a path_delete, and that'd produce a different action.